src/Integration/LocalServices/SolutionRuleSetsInformationProvider.cs

```diff
@@ -46,22 +46,38 @@ public SolutionRuleSetsInformationProvider(IServiceProvider serviceProvider)
 
         public IEnumerable<RuleSetDeclaration> GetProjectRuleSetsDeclarations(Project project)
         {
+            /* This method walks through all of the available configurations (e.g. Debug, Release, Foo) and
+             * attempts to fetch the values of a couple of properties from the project (CodeAnalysisRuleSet
+             * and CodeAnalysisRuleSetDirectories). The collected data is put into a data object
+             * and returned to the caller. The collected data includes the DTE Property object itself, which
+             * is used later to update the ruleset value.
+             * 
+             * TODO: consider refactoring. The code seems over-complicated: it finds the "ruleset"
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)